src/app/frontend/deploy/deployfromsettings.html

```diff
@@ -17,17 +17,19 @@
 <kd-help-section>
   <md-input-container class="md-block" md-is-error="ctrl.isNameError()">
     <label>App name</label>
-    <input ng-model="ctrl.name" name="name" namespace="ctrl.namespace" required kd-unique-name
-        ng-model-options="{ updateOn: 'default blur', debounce: { 'default': 500, 'blur': 0 } }">
-    <md-progress-linear class="kd-deploy-form-progress" md-mode="indeterminate"
-        ng-show="ctrl.form.name.$pending">
-    </md-progress-linear>
-    <ng-messages for="ctrl.form.name.$error" role="alert" multiple>
-      <ng-message when="required">Application name is required.</ng-message>
-      <ng-message when="uniqueName">
-        Application with this name already exists within namespace <i>{{ctrl.namespace}}</i>.
-      </ng-message>
-    </ng-messages>
+    <input ng-model="ctrl.name"
+           name="name"
```

Comment:
There are some side-effects to having `ng-model-options`, and side effects to not having it. I'll enumerate the negative impact with and without here for clarity: 
WITH: the progress bar only occasionally shows for a fraction of a second, and most of the time it doesn't show at all. This is because it only starts showing from the moment the 'name' is permitted to update, removing a minimum of 500 ms from its total showing time which is most often less than that.
WITH: validation appears laggy to the user who can sometimes get a 'app must have a name' error message still showing for 1/2 second after he/she has been imputing valid characters, or a 'wrong characters' message for 1/2 second after his/her input is valid etc. 1/2 is a long time in terms of UX.
WITHOUT: calls are made to the server with every keystroke to check for unique name.
without: the aforementioned can cause validation (for uniqueName) to take longer while the calls to the server are made, but the progress bar shows during the entire time of that process.
WITHOUT: the name in the help section you mentioned doesn't flicker, but does update to reflect the user's input as soon as each validation has occured which can be on every keystroke if they are slow enough.
I have tried implementing a solution which shows the name in the help section only when the 'name' input does not have focus, but when the user changes focus the name has not always finished validating, causing the old name to show until validation (slow in this case) has finished, at which point it updates. 
@bryk @maciaszczykm @cheld, which solution should I implement?